lib/Resque/Job.php

```diff
@@ -230,7 +230,12 @@ public function fail($exception)
 			$this->worker,
 			$this->queue
 		);
-		Resque_Stat::incr('failed');
+
+        if (Resque_Failure::getBackend() == 'Resque_Failure_RedisMultipleQueues') {
+            Resque_Stat::incr($this->queue . '_failed');
+        } else {
+            Resque_Stat::incr('failed');
+        }
```

Comment:
Pushing the stat increment up into the failure object itself is definitely an option.  I hesitated to suggest that because taking a quick look at the code it kind of feels like that's not the philosophy here - it looks like the ``Resque_Job`` is taking responsibility for stats, that's only with a cursory glance though.

If there's no killer reason that we need the stat to simply say 'failed' for non-multiple queued failure backends then I'd say we should keep it simply and simply include the queue as part of the stat by default.

If there's a reason we can't have that, or we simply aren't sure, then pushing the responsibility up to the failure instance is reasonable.

But I definitely think we want to avoid making this method need to know about specific failure object implementations.